hibernate-reactive-core/build.gradle

```diff
@@ -26,6 +26,9 @@ dependencies {
 
     //Specific implementation details of Hibernate Reactive:
     implementation "io.vertx:vertx-sql-client:${vertxVersion}"
+    implementation 'io.agroal:agroal-api:1.12'
+    implementation 'io.agroal:agroal-pool:1.12'
+    implementation "io.vertx:vertx-jdbc-client:${vertxVersion}"
```

Comment:
We now have to figure out how to move everything in a separate optional module. People that aren't using H2 won't need this libraries